src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -430,4 +454,80 @@ private teammates.storage.sqlentity.FeedbackResponseComment createFeedbackRespon
         return newComment;
     }
 
+    // methods for misc migration methods ---------------------------------------------------------------------------------
+
+    private void migrateInstructorEntities(teammates.storage.sqlentity.Course newCourse,
+            Map<String, User> userEmailToUserMap) {
+        List<Instructor> oldInstructors = ofy().load().type(Instructor.class).filter("courseId", newCourse.getId())
+                .list();
+        for (Instructor oldInstructor : oldInstructors) {
+            teammates.storage.sqlentity.Instructor newInstructor = migrateInstructor(newCourse, oldInstructor);
+            userEmailToUserMap.put(oldInstructor.getEmail(), newInstructor);
+        }
+    }
+
+    private teammates.storage.sqlentity.Instructor migrateInstructor(teammates.storage.sqlentity.Course newCourse,
+            Instructor oldInstructor) {
+        InstructorPrivileges newPrivileges;
+        if (oldInstructor.getInstructorPrivilegesAsText() == null) {
+            newPrivileges = new InstructorPrivileges(oldInstructor.getRole());
+        } else {
+            InstructorPrivilegesLegacy privilegesLegacy = JsonUtils
+                    .fromJson(oldInstructor.getInstructorPrivilegesAsText(), InstructorPrivilegesLegacy.class);
+            newPrivileges = new InstructorPrivileges(privilegesLegacy);
+        }
+
+        teammates.storage.sqlentity.Instructor newInstructor = new teammates.storage.sqlentity.Instructor(
```

Comment:
Does the instructor need an account and team also as a user? 

Accounts have already been migrated, just need to fetch them when making instructors and students. If you think the change is too big, can open another PR to include accounts, or you can do it in this PR too

Reply:
Done in #13073